src/library/scala/collection/immutable/StringLike.scala

```diff
@@ -203,8 +203,28 @@ self =>
 
   private def escape(ch: Char): String = "\\Q" + ch + "\\E"
 
-  @throws(classOf[java.util.regex.PatternSyntaxException])
-  def split(separator: Char): Array[String] = toString.split(escape(separator))
```

Comment:
What is wrong with deferring to the Java single-character `split`?  It doesn't do regex matching so you don't need to quote it.